README.txt
p4a.video
=========

Overview
--------

The *p4a.video* package provides a framework for handling video content on
the Zope 2 and Zope 3 platforms.  It was inspired by the Plone ATVideo
product and even borrows some UI.

Project Description
-------------------
*p4a.video* allows you to:

* convert a file object to a video
* extract the video metadata
* render an appropriate view (depending on the file format)

Installation
------------
*p4a.video* depends on the following packages:

* p4a.subtyper
* p4a.common
* p4a.z2utils
* p4a.fileimage
* hachoir_core
* hachoir_metadata
* hachoir_parser

If you use buildout, you can have it manage these dependencies by:

* adding these lines to your buildout.cfg file

::

    [buildout]
    ...
    
    eggs = 
        ...
        p4a.video
    
    [instance]
    ...
    
    zcml = 
        ...
        p4a.video

* re-run your buildout
* Install the add-ons from the Add-on Products page in the
  Plone Control Panel

Features
--------

Video Files
~~~~~~~~~~~
* Support for:

  * Quicktime (MOV, MP4)
  * Windows Media (WMV, AVI, WMA, ASF)
  * RealMedia (RAM)
  * Flash (FLV, SWF)

* Upload a thumbnail image to represent the video
* When clicked, the video will start to play video
* edit form exposes metadata:

  * file type
  * author
  * height / width
  * duration
 
Video containers
~~~~~~~~~~~~~~~~
* Turn any folder into a video container
* Turn any smart folder (collection) into a video container
* Provides a video listing view with all videos in the folder, including:

  * thumbnail
  * title
  * description
  * metadata

* Video listing also shows:

  * tags
  * ratings
  * comments about each video

Video feeds
~~~~~~~~~~~
* Publish RSS feed of all the videos in a video container
* Feed entries can contain a link to the video view page, or a link to the
  actual video file (enclosure)
* Users can subscribe to a vodcast and have the videos downloaded to a
  desktop video player such as iTunes or Miro for offline viewing

Video feedback / commentary
~~~~~~~~~~~~~~~~~~~~~~~~~~~
* Requires:

  * lovely.tag
  * p4a.plonetagging

* Users can rate videos (1 to 5 stars)

  * Plone keeps track of user ratings vs. editor ratings

* Users can tag videos

  * Plone keeps track of your tags and everyone's tags

* Users can comment on videos. Other users can reply to those comments in a
  threaded discussion


docs/CHANGES.txt
Changes
=======

1.4 (XXXX-XX-XX)
----------------
- Made VideoPageView inherit also from BrowserView so that it can be customized
  ttw with portal_view_customizations [ichimdav]
- Fix IE conflict with kukit
  [afd]

- Included the zcml from p4a.subtyper.
  [davidblewett]

- Included the zcml from p4a.fileimage, so it is not necessary to include
  manually when adding p4a.video to a buildout.
  [davidblewett]

- Minimum Plone version supported is now 3.2, removed all backwards-compatibility
  support for Plone 2.5.
  [davidblewett]

- Updated events to import from zope.lifecycleevent, instead of zope.app.event.
  [davidblewett]

- Changed annotation support to use zope.annotation instead of zope.app.annotation.
  [davidblewett]

1.3 (2010-07-28)
----------------

- Updated flowplayer to version 3. This allows it to work on IE again.
  Thanks to cdw9 for the help with the template.
  [claytron]

1.2 (2010-06-18)
----------------

- Added CSS and JS files to their respective registries.
  Removed them from video.pt.
  [davidblewett]

- Updated README to include dependency information in the Install section.
  [davidblewett]

- Updated README with information from the plone4artistsvideo product page.
  Changed setuptools' long description to use the contents of README.txt.
  [davidblewett]

- Updated old reference to p4acal.
  [davidblewett]

- Forgot to commit more translations. Sorry
  [tzicatl]

- Added more translation strings.
  Added Spanish translation strings
  [tzicatl]

- Remove some unused imports
  [rosspatterson]

- Pyflakes fix
  [rosspatterson]

- add the it translation
  [stemrc]

- pinned Hachoir versions to 1.2 since the latest 1.2.1 only works with Python 2.5
  [nate]


1.1.1 (2008-10-10)
------------------

Features
~~~~~~~~

* Support for iPod video - video/x-m4v.  TODO: tests.  (Nate Aune)

* Brazilian Portuguese translation. (ericoandrei)

* Preliminary fixed rate Real media support. TODO:
  tests. (danielblackburn)

Bug Fixes
~~~~~~~~~

* Fix rich text widget handling.  TODO: tests.  (Nate Aune)

* Depend on hachoir. (Nate Aune)

* fixed issue where url assignment, was not updated properly with rest
  of the method. TODO: tests. (danielblackburn)

1.1 (2008-04-09)
----------------

Bug Fixes
~~~~~~~~~

* Fixed ContentLicensing optional'ness to fix issue #40 (Ross Patterson and
  Rocky Burt)

* Fixed handling of unicode AT titles for issues #41 and #47 (Ross Patterson
  and Rocky Burt)

* Updated vPIP (Nate Aune)

* Updated flowplayer (Ross Patterson and Rocky Burt)

* Portal status message handling was made to work better to fix issue #38
  (Ross Patterson)

* Hooked up video/mpeg mime type and player link to fix issue #28
  (Ross Patterson)

1.1b1 (2008-02-13)
------------------

Features
~~~~~~~~

* Added Plone 3 compatibility (Rocky Burt) r1794

* Added rich text description field (Rocky Burt) r1829

* Now uses p4a.subtyper to expose video subtype action (Rocky Burt) r1787

Bug Fixes
~~~~~~~~~

* Support additional mime type synonyms - video/x-msvideo and video/x-flash-video (Ross Patterson) r1881

* Support reasonable default dimensions for the players
  (Ross Patterson) r1920

Miscellaneous
~~~~~~~~~~~~~

* Now uses standard plone/cmf comments instead of easycommenting
  (Rocky Burt)


1.0 (2007-11-28)
----------------

Features
~~~~~~~~

* Default video container view has been updated (Rocky Burt)

* Added description field to IVideo and subsequent views (Rocky Burt)


1.0beta1 (2007-07-31)
---------------------

* Various bug fixes and general improvments.


1.0alpha1 (2007-04-29)
----------------------

* First release.



p4a/video/genre.py
from zope.schema import vocabulary

GENRES = (
    (0, "Blues"),
    (1, "Classic Rock"),
    (2, "Country"),
    (3, "Dance"),
    (4, "Disco"),
    (5, "Funk"),
    (6, "Grunge"),
    (7, "Hip-Hop"),
    (8, "Jazz"),
    (9, "Metal"),
    (10, "New Age"),
    (11, "Oldies"),
    (12, "Other"),
    (13, "Pop"),
    (14, "R&B"),
    (15, "Rap"),
    (16, "Reggae"),
    (17, "Rock"),
    (18, "Techno"),
    (19, "Industrial"),
    (20, "Alternative"),
    (21, "Ska"),
    (22, "Death Metal"),
    (23, "Pranks"),
    (24, "Soundtrack"),
    (25, "Euro-Techno"),
    (26, "Ambient"),
    (27, "Trip-Hop"),
    (28, "Vocal"),
    (29, "Jazz+Funk"),
    (30, "Fusion"),
    (31, "Trance"),
    (32, "Classical"),
    (33, "Instrumental"),
    (34, "Acid"),
    (35, "House"),
    (36, "Game"),
    (37, "Sound Clip"),
    (38, "Gospel"),
    (39, "Noise"),
    (40, "AlternRock"),
    (41, "Bass"),
    (42, "Soul"),
    (43, "Punk"),
    (44, "Space"),
    (45, "Meditative"),
    (46, "Instrumental Pop"),
    (47, "Instrumental Rock"),
    (48, "Ethnic"),
    (49, "Gothic"),
    (50, "Darkwave"),
    (51, "Techno-Industrial"),
    (52, "Electronic"),
    (53, "Pop-Folk"),
    (54, "Eurodance"),
    (55, "Dream"),
    (56, "Southern Rock"),
    (57, "Comedy"),
    (58, "Cult"),
    (59, "Gangsta"),
    (60, "Top 40"),
    (61, "Christian Rap"),
    (62, "Pop/Funk"),
    (63, "Jungle"),
    (64, "Native American"),
    (65, "Cabaret"),
    (66, "New Wave"),
    (67, "Psychadelic"),
    (68, "Rave"),
    (69, "Showtunes"),
    (70, "Trailer"),
    (71, "Lo-Fi"),
    (72, "Tribal"),
    (73, "Acid Punk"),
    (74, "Acid Jazz"),
    (75, "Polka"),
    (76, "Retro"),
    (77, "Musical"),
    (78, "Rock & Roll"),
    (79, "Hard Rock"),
      
# Winamp genres      
    (80, "Folk"),
    (81, "Folk-Rock"),
    (82, "National Folk"),
    (83, "Swing"),
    (84, "Fast Fusion"),
    (85, "Bebob"),
    (86, "Latin"),
    (87, "Revival"),
    (88, "Celtic"),
    (89, "Bluegrass"),
    (90, "Avantgarde"),
    (91, "Gothic Rock"),
    (92, "Progressive Rock"),
    (93, "Psychedelic Rock"),
    (94, "Symphonic Rock"),
    (95, "Slow Rock"),
    (96, "Big Band"),
    (97, "Chorus"),
    (98, "Easy Listening"),
    (99, "Acoustic"),
    (100, "Humour"),
    (101, "Speech"),
    (102, "Chanson"),
    (103, "Opera"),
    (104, "Chamber Music"),
    (105, "Sonata"),
    (106, "Symphony"),
    (107, "Booty Bass"),
    (108, "Primus"),
    (109, "Porn Groove"),
    (110, "Satire"),
    (111, "Slow Jam"),
    (112, "Club"),
    (113, "Tango"),
    (114, "Samba"),
    (115, "Folklore"),
    (116, "Ballad"),
    (117, "Power Ballad"),
    (118, "Rhythmic Soul"),
    (119, "Freestyle"),
    (120, "Duet"),
    (121, "Punk Rock"),
    (122, "Drum Solo"),
    (123, "A capella"),
    (124, "Euro-House"),
    (125, "Dance Hall"),
    (126, "Goa"),
    (127, "Drum & Bass"),
    (128, "Club - House"),
    (129, "Hardcore"),
    (130, "Terror"),
    (131, "Indie"),
    (132, "BritPop"),
    (133, "Negerpunk"),
    (134, "Polsk Punk"),
    (135, "Beat"),
    (136, "Christian Gangsta Rap"),
    (137, "Heavy Metal"),
    (138, "Black Metal"),
    (139, "Crossover"),
    (140, "Contemporary Christian"),
    (141, "Christian Rock"),
    (142, "Merengue"),
    (143, "Salsa"),
    (144, "Thrash Metal"),
    (145, "Anime"),
    (146, "JPop"),
    (147, "Synthpop"),     
 )

_terms = [vocabulary.SimpleTerm(key, key, value) for key, value in GENRES]
GENRE_VOCABULARY = vocabulary.SimpleVocabulary(_terms)
